### PR TITLE
Update LastMet and NextMeeting validity checks

### PR DESCRIPTION
**Pull request by benedictchuajj**
### Description

LastMet now has checks that prevents users from inputting future dates (which does not make sense and is a potential featureFlaw)

Also, updates isValidLastMet as well as isValidNextMeeting that was missed out in #198 

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### Checklist

- [X] I have tested this code
- [ ] I have updated the documentation

### CODE REVIEW

**Comment by benedictchuajj** [comment]
@KT27Learn This might fix the NextMeeting problem in your PR (#209). PS

File: src/main/java/seedu/address/model/client/LastMet.java [hunk 1]
```diff
@@ -44,7 +46,18 @@ public LastMet(String lastMetDate) {
...
      */
     public static boolean isValidLastMet(String test) {
-        return StringUtil.isValidDate(test);
+        return StringUtil.isValidDate(test) && notFutureDate(test);
+    }
+
+    /**
+     * Returns if a given string contains a date that is not in the future.
+     */
+    public static boolean notFutureDate(String test) {
```
Comment (CrownKira) [inline src/main/java/seedu/address/model/client/LastMet.java:55]: Perhaps `isNotFutureDate` would better

Comment (benedictchuajj) [reply src/main/java/seedu/address/model/client/LastMet.java]: Thanks for the suggestion!

**Review by wyrchris — COMMENTED** [review]
Looks goods, work as expected. However, it seems the intended message "LastMet should not be a future date." does not get displayed when a future date is used.

Also, possibly on a separate issue, next meeting and last meeting respective parserUtil methods check is not as strict as that of the check done in the constructor method. This result in the check which pass in the parserUtil method but fail in the constructor method result in IllegalArgumentException/RuntimeException being thrown as opposed to the expected ParserException which does not get catch. Not sure whether this is considered an issue

**Comment by benedictchuajj** [comment]
I have updated the respective ParserUtil methods to do more stringent checks and to utilise the correct error message to direct the user to which part of their command is wrong

**Review by wyrchris — APPROVED** [review]
Looks good! Feel free to merge once the merge conflict is resolved.

**Pull request merged**